src/Network/Wai/SAML2/Request.hs

```diff
@@ -0,0 +1,53 @@
+module Network.Wai.SAML2.Request
+    ( Request(..)
+    , issueRequest
+    ) where
+
+import Crypto.Random
+import Data.Time.Clock
+import Data.Time.Format
+import Network.Wai.SAML2.XML
+import Text.XML
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Base64 as Base64
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+
+-- | Parameters for SP-initiated SSO
+data Request = Request
+  { requestIssuer :: T.Text
+  , requestAllowCreate :: Bool
+  , requestNameIDFormat :: T.Text
+  }
```

Comment:
`Request` is a pretty generic name. I am wondering if it might be better to rename this to e.g. `AuthnRequest`, which would also better correspond to the XML element name?

There is also no documentation for the fields here. I won't hold up the PR because of this, but it would be nice. Again, see the [contribution guidelines](https://github.com/mbg/wai-saml2/blob/master/CONTRIBUTING.md#documentation) with respect to documentation. In particular, it might also be nice to extend the README with documentation on how to use `issueRequest`, or provide module-level documentation.

Reply:
I renamed the datatype and added comments to each field